CMakeLists.txt

```diff
@@ -1,144 +1,78 @@
-endif()
+include(GNUInstallDirs)
+include(cmake/SetDefaultProfile.cmake)
+include(cmake/CompilerOptions.cmake)
+include(cmake/Utils.cmake)
+# For custom Find*.cmake modules
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/modules")
 
 # Output all binaries in the same directory for easier testing
 string(TOUPPER ${CMAKE_BUILD_TYPE} BUILD_TYPE)
 string(TOLOWER ${CMAKE_SYSTEM_NAME} OPERATING_SYSTEM)
 string(TOLOWER ${CMAKE_CXX_COMPILER_ID} COMPILER)
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${BUILD_TYPE} "${CMAKE_BINARY_DIR}/bin/${OPERATING_SYSTEM}-${CMAKE_SYSTEM_PROCESSOR}-${COMPILER}-${CMAKE_BUILD_TYPE}")
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_${BUILD_TYPE} "${CMAKE_BINARY_DIR}/bin/${OPERATING_SYSTEM}-${CMAKE_SYSTEM_PROCESSOR}-${COMPILER}-${CMAKE_BUILD_TYPE}")
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_${BUILD_TYPE} "${CMAKE_BINARY_DIR}/bin/${OPERATING_SYSTEM}-${CMAKE_SYSTEM_PROCESSOR}-${COMPILER}-${CMAKE_BUILD_TYPE}")
+set(OUTPUT_DIR "${CMAKE_BINARY_DIR}/bin/${OPERATING_SYSTEM}-${CMAKE_SYSTEM_PROCESSOR}-${COMPILER}-${CMAKE_BUILD_TYPE}")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${BUILD_TYPE} "${OUTPUT_DIR}")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_${BUILD_TYPE} "${OUTPUT_DIR}")
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_${BUILD_TYPE} "${OUTPUT_DIR}")
 
-# Find Git package, if not need to install manually or through .yml file
-find_package(Git)
-if(NOT Git_FOUND)
-    message(FATAL_ERROR "Git was not found. Install Git and make sure it is in your PATH.")
-endif()
+# Look for shared libs in the same directory as the executable when running tests
+set(CMAKE_BUILD_RPATH "\$ORIGIN")
 
+find_package(Git REQUIRED)
 if(NOT DEFINED GIT_BRANCH)
     set(GIT_BRANCH "main")
 endif()
-
-# Build version of EDIE through cmake
-if(GIT_EXECUTABLE)
-    execute_process(COMMAND ${CMAKE_COMMAND}
-        -D SRC=${CMAKE_CURRENT_SOURCE_DIR}/src/version.h.in
-        -D DST=${CMAKE_CURRENT_SOURCE_DIR}/src/version.h
-        -D GIT_EXECUTABLE=${GIT_EXECUTABLE}
-        -D GIT_BRANCH=${GIT_BRANCH}
-        -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/GenerateVersionHeader.cmake)
-endif()
-
-include_directories(
-    ${CMAKE_CURRENT_SOURCE_DIR}
-    ${CMAKE_CURRENT_SOURCE_DIR}/src
-)
+execute_process(COMMAND ${CMAKE_COMMAND}
+    -D SRC=${CMAKE_CURRENT_SOURCE_DIR}/src/version.h.in
+    -D DST=${CMAKE_CURRENT_SOURCE_DIR}/include/novatel_edie/version.h
+    -D GIT_EXECUTABLE=${GIT_EXECUTABLE}
+    -D GIT_BRANCH=${GIT_BRANCH}
+    -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/GenerateVersionHeader.cmake)
 
 # Build EDIE components
-add_subdirectory(src/hw_interface)
-add_subdirectory(src/decoders)
+add_subdirectory(src/common)
+add_subdirectory(src/stream_interface)
+add_subdirectory(src/decoders/common)
+add_subdirectory(src/decoders/oem)
+
+# Add an aggregate target for all components
+add_library(novatel_edie INTERFACE)
+add_library(novatel_edie::novatel_edie ALIAS novatel_edie)
+target_link_libraries(novatel_edie INTERFACE
+    common
+    stream_interface
+    decoders_common
+    oem_decoder
```

Comment:
shouldn't these libraries should have the same naming format?

Reply:
`common_decoder` sounds like a component that is capable of decoding something on its own rather than providing base classes/functionality, no? Or is that the case?

`decoders_oem` is an option. What do you think?

Reply:
I see the logic in keeping it the way it is